model.js

```diff
@@ -48,3 +48,25 @@ exports.returnReviewComments = (reviewId) => {
     return Promise.reject({ status: 400, msg: err });
   }
 };
+
+exports.postNewComment = (params, reviewId) => {
+  if (typeof +reviewId === 'number') {
+    return db.query(`SELECT * FROM reviews;`).then((response) => {
+      let reviewIds = response.rows.map((x) => x.review_id);
+      if (reviewId > Math.max(...reviewIds)) {
+        return Promise.reject({ status: 404, msg: 'Not found' });
+      } else {
+        return db
+          .query(
+            `INSERT INTO comments (body, author, review_id) VALUES ($1, $2, $3) RETURNING *;`,
+            [params.body, params.username, reviewId]
+          )
+          .then((comment) => {
+            return comment.rows;
+          });
+      }
+    });
+  } else {
+    return Promise.reject({ status: 400, msg: err });
```

Comment:
The review ID type being the wrong datatype will result in a PSQL error, which should be caught in app.js - checking the error code (which will be something like '22P02'). The advantage of catching the error there rather than in the model, is that future endpoints can be caught in the same way, rather than having to put an if statement checking the datatype in several model functions, like you have on line 53.